collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java

```diff
@@ -200,30 +217,51 @@ public MapView getMapView() {
         longPressListener = listener;
     }
 
+    @Override public void setGpsLocationListener(@Nullable PointListener listener) {
+        gpsLocationListener = listener;
+    }
+
     @Override public void runOnGpsLocationReady(@NonNull ReadyListener listener) {
         myLocationOverlay.runOnFirstFix(() -> getActivity().runOnUiThread(() -> listener.onReady(this)));
     }
 
-    @Override public void setGpsLocationEnabled(boolean enabled) {
-        if (enabled) {
-            LocationManager locationManager = (LocationManager) getContext().getSystemService(Context.LOCATION_SERVICE);
-            if (locationManager != null && locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
-                map.getOverlays().add(myLocationOverlay);
-                myLocationOverlay.setEnabled(true);
-                myLocationOverlay.enableMyLocation();
+    @Override public void setGpsLocationEnabled(boolean enable) {
```

Comment:
In the idealized spec, I wrote 
> Google Maps gets current location if only network location is enabled. OSM requires GPS to be turned on. In Android 5.1.1, Privacy > Location > Locating method > Wi-Fi and mobile networks sets that up.

It looks like requiring GPS rather than just any available provider was a very deliberate decision here. We should discuss at some point whether it's still the best behavior. My sense is that it should be consistent with Google Maps.

Reply:
There's a user-facing impact here.

Ramani Huria needs/wants the GeoTrace activity to require GPS because they want their traces to be accurate; when location comes from a non-GPS provider, it's so inaccurate as to be useless for them.  It's important to them that the app forces users to turn on GPS so that they don't inadvertently collect bad data.

It seems like, in the ideal case, requiring or not requiring GPS would be another option in the form definition; does that sound right to you?

As for right now, in OSM mode, there's a check for specifically the GPS provider and in Google Maps mode, any location provider is accepted.  This is all the same as before, as far as I can tell.  So the behaviour hasn't changed, though in Google Maps mode, the name of the method `showGpsDisabledAlert` and the text that is shown in the alert dialog are misleading.

Reply:
> in Google Maps mode, the name of the method showGpsDisabledAlert and the text that is shown in the alert dialog are misleading. Should I do something about that before phase 3?

I don't think so.

Let's come back to the broader provider question. We'll also need to figure out what to do when Google fixes https://github.com/opendatakit/collect/issues/2008 in the next Play Services release. I've made a note of it at the bottom of phase 2 [here](https://docs.google.com/document/d/1r3wJrOVDeFQkk4kWMoW29LtLI1vbBChqJ65w1Jyg4YU/edit).